src/components/EditProfile.vue

```diff
@@ -170,71 +171,70 @@ export default {
     },
     saveImage: function () {
       // Save the uploaded profile picture
-      axios({
+      return axios({
         method: 'put',
-        url: this.updateUser.url,
+        url: this.updateUser.postUrl,
         data: this.profileImage
       })
-        .then(res => {
-        })
-        .catch(error => {
-          console.log(error)
-        })
+    },
+    getProfile: function () {
+      return axios({
+        method: 'get',
+        url: '/api/users/' + this.getCurrentUser,
+        headers: {
+          'Authorization': 'Bearer ' + this.$store.state.token
+        }
+      })
+    },
+    updateProfile: function () {
+      return axios({
+        method: 'put',
+        url: '/api/users/edit/' + this.getCurrentUser,
+        headers: {
+          'Authorization': 'Bearer ' + this.$store.state.token
+        },
+        data: this.updateUser
+      })
     },
     saveProfileSettings () {
       this.userProfile = this.cloneUserProfile
       // Makes sure to set up data object with all data needed for vuex call
       // *Honestly this is ridiculous and needs improvement, but it works for now*
+      this.updateUser.postUrl = this.cloneUserProfile.postUrl
       this.updateUser.id = this.getCurrentUser
       this.updateUser.firstname = this.cloneUserProfile.firstname
       this.updateUser.lastname = this.cloneUserProfile.lastname
       this.updateUser.picid = this.cloneUserProfile.picid
-      this.updateUser.url = this.cloneUserProfile.url
       this.updateUser.email = this.cloneUserProfile.email
       this.updateUser.phone = this.cloneUserProfile.phone
       this.updateUser.items = this.cloneUserProfile.items
       this.updateUser.zipcode = this.cloneUserProfile.zipcode
-      this.saveImage()
+      let promises = []
+      promises.push(this.saveImage())
       this.updateUser.url = null
       if (this.validateEmail(this.userProfile.email)) {
         // Make API call to update the user info and refresh data on front-end
-        axios({
-          method: 'put',
-          url: '/api/users/edit/' + this.getCurrentUser,
-          headers: {
-            'Authorization': 'Bearer ' + this.$store.state.token
-          },
-          data: this.updateUser
-        })
+        promises.push(this.updateProfile())
+      }
+      Promise.all(promises).then(res => {
```

Comment:
I think for now it will be fine, definitely need to come back to this. If the update fails with 401, the GetUser will fail with 401 immediately afterwards and redirect. Not the cleanest, but effective enough (to me) for the beta